src/lib/device/device.h

```diff
@@ -15,204 +14,167 @@
+    uint8_t failed{0};                         // 25: set to true if disk is replaced
+    uint8_t hs_dev_type{0};                    // 26: PDev dev type (as in fast or data)
+    uint8_t multi_pdev_choice{0};              // 27: Choice when multiple pdevs are present (vdev_multi_pdev_opts_t)
+    char name[64];                             // 28: Name of the vdev
+    uint16_t checksum{0};                      // 94: Checksum of this entire block
+    uint8_t padding[32]{};                     // 96: Pad to make it 256 bytes total
+    uint8_t user_private[user_private_size]{}; // 128: User specific information
+
+    uint32_t get_vdev_id() const { return vdev_id; }
+    uint64_t get_size() const { return vdev_size; }
+
+    void set_name(const std::string& n) { std::strncpy(charptr_cast(name), n.c_str(), 63); }
+    std::string get_name() const { return std::string{c_charptr_cast(name)}; }
+
+    void set_allocated() { slot_allocated = s_cast< uint8_t >(0x01); };
+    void set_free() { slot_allocated = s_cast< uint8_t >(0x00); }
+    bool is_allocated() const { return (slot_allocated == 0x01); }
+
+    void mark_failed() { failed = s_cast< uint8_t >(0x01); }
+    void unfail() { failed = s_cast< uint8_t >(0x00); }
+    bool is_failed() const { return (failed == 0x01); }
+
+    void set_dev_type(HSDevType dtype) { hs_dev_type = enum_value(dtype); }
+    void set_pdev_choice(vdev_multi_pdev_opts_t opts) { multi_pdev_choice = enum_value(opts); }
+
+    void set_user_private(const sisl::blob& data) {
+        std::memcpy(&user_private, data.bytes, std::min(data.size, uint32_cast(user_private_size)));
+    }
+    uint8_t* get_user_private_mutable() { return &(user_private[0]); }
+    const uint8_t* get_user_private() const { return &(user_private[0]); }
+
+    void compute_checksum() {
+        checksum = 0;
+        checksum = crc16_t10dif(hs_init_crc_16, r_cast< const unsigned char* >(this), sizeof(vdev_info));
+    }
+};
+#pragma pack()
+
+static_assert(sizeof(vdev_info) <= vdev_info::size, "VDev info sizeof() mismatch");
+
+struct vdev_parameters {
+    std::string vdev_name;                  // Name of the vdev
+    uint64_t vdev_size;                     // Current Vdev size.
+    uint32_t num_chunks;                    // Total number of primary chunks.
+                                            // NOTE: If pdev opts is ALL_PDEV_STRIPED, then num_chunks would round off
+                                            // to number of pdevs evenly
+    uint32_t blk_size;                      // Block size vdev operates on
+    HSDevType dev_type;                     // Which physical device type this vdev belongs to (FAST or DATA)
+    vdev_multi_pdev_opts_t multi_pdev_opts; // How data to be placed on multiple vdevs
+    sisl::blob context_data;                // Context data about this vdev
+};
+
+ENUM(chunk_selector_t, uint8_t, // What are the options to select chunk to allocate a block
```

Comment:
I believe this is obsoleted by chunk_selector_type_t defined in homestore_decl.hpp, right?

Reply:
I moved this one to homestore_decl.hpp. Thanks for pointing it out.